### PR TITLE
Add split string function

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

**Comment by haozhun** [comment]
I believe I forgot submitting pull request for this one until I saw your post.

File: presto-docs/src/main/sphinx/functions/string.rst [hunk 1]
```diff
@@ -56,6 +56,16 @@ String Functions
 
     Removes trailing spaces from ``string``.
 
+.. function:: split(string, delimiter) -> array<varchar>
+
+    Splits ``string`` on ``delimiter`` and return an array.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/string.rst]: should say "returns an array"

File: presto-docs/src/main/sphinx/functions/string.rst [hunk 2]
```diff
@@ -56,6 +56,16 @@ String Functions
...
+
+.. function:: split(string, delimiter, limit) -> array<varchar>
+
+    Splits ``string`` on ``delimiter`` and return an array of size at most
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/string.rst]: should say "returns an array"

File: presto-docs/src/main/sphinx/functions/string.rst [hunk 3]
```diff
@@ -56,6 +56,16 @@ String Functions
...
+    ``limit``. The last element in the array always contain everything
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/string.rst]: Hmm, this seems weird to put everything in the last element. Check with @electrum or @martint if that's how it should work

File: presto-docs/src/main/sphinx/functions/string.rst [hunk 4]
```diff
@@ -56,6 +56,16 @@ String Functions
...
+    left in the ``string``. If ``limit`` is zero, no limit is specified.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/string.rst]: Limit being zero should be an error, imo. @electrum, @martint thoughts?

**Comment by haozhun** [comment]
@electrum @martint  Github incorrectly considers some of @cberner's comments as outdated. Let me repeat the question:

* What should be the last element of the result array of `split('a.a.a', '.', 2)`?
* What should happen if user calls `split('a.a.a', '.', 0)`? Consider it an error or equivalent to `split('a.a.a', '.')`?

File: presto-docs/src/main/sphinx/functions/string.rst (hunk 3 above)
Comment (martint) [inline presto-docs/src/main/sphinx/functions/string.rst]: Actually, that sounds like reasonable behavior. It's what most other languages do (ruby, java, etc).

**Comment by martint** [comment]
Given that `limit` indicates how many items the array will contain at most, limit 1 should be equivalent to calling the variant without limit, and limit 0 should be an error (worst case, it should return an empty array, but that seems pretty useless)

**Comment by cberner** [comment]
Works for me, and the implementation looked correct. @electrum any comments on the function name?

**Comment by haozhun** [comment]
@martint Argument 1 is note equivalent to calling without limit. Infinity is. I'll make zero an error.

**Comment by martint** [comment]
Ah, yes, you're right. I was thinking of the case where limit = 1 gives you an array with a single entry containing the entire string.

**Comment by haozhun** [comment]
Ready for review again

**Comment by martint** [comment]
looks good

**Pull request merged**